src/utils/spark.py

```diff
@@ -55,6 +65,12 @@ def __init__(self, app_name: str, password_file_path: str) -> None:
             f"{self.ipts_service_name}"
         )
 
+        # GitHub credentials for launching a target workflow once all jobs
+        # are finished running
+        self.gh_app_id = os.getenv("GH_APP_ID")
+        self.gh_repo = os.getenv("GH_REPO")
+        self.gh_workflow = os.getenv("GH_WORKFLOW")
```

Comment:
**[Question, non-blocking]** What's the intuition for the distinction between variables that are loaded from the environment vs. from the arguments passed to the constructor method? I find it a little bit confusing that we read class attributes from both places, but I could be convinced that it's fine if there's a good reason to do so. Otherwise, my preference would be for the classes to accept all of their attributes via constructor params, such that the responsibility falls on the calling code in `main()` to determine where to load these values.

Reply:
The distinction I've made so far is that anything that loads from a file is loaded via argument, while anything that loads from the environment is read on init. I mainly didn't want classes with a billion constructor params, but happy to refactor if you feel strongly!

Reply:
Eh, let's leave it for now, and we can easily refactor later if it becomes a pain!